File: OptiPlanBackend/OptiPlanBackend/Models/Invitation.cs
using OptiPlanBackend.Enums;
using System.ComponentModel.DataAnnotations;

namespace OptiPlanBackend.Models
{
    public class Invitation
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid TeamId { get; set; }
        public Team Team { get; set; }

        // Invited user (if registered)
        public Guid? InviteeId { get; set; }
        public User? Invitee { get; set; }

        // Email for non-registered users
        [Required]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        // Invitation metadata
        public InvitationStatus Status { get; set; } = InvitationStatus.Pending;
        public DateTime SentAt { get; set; } = DateTime.UtcNow;
        public DateTime? RespondedAt { get; set; }

        // Inviter (who sent the invite)
        [Required]
        public Guid InviterId { get; set; }
        public User Inviter { get; set; }

        // Old property replaced by Status enum
        [Obsolete("Use Status property instead")]
        public bool IsAccepted { get; set; } = false;
    }
}

File: optiplan-frontend/src/app/models/invitation.ts

import { InvitationStatus } from './enums/invitation-status';
import { TeamRole } from './enums/team-role';
import { Team } from './team';
import { User } from './user';

export interface Invitation {
  id: string;
  teamId: string;
  team: Team;

  // Invited user (nullable in C#)
  inviteeId?: string;
  invitee?: User;

  role: TeamRole;

  email: string;

  status: InvitationStatus;
  sentAt: string;
  respondedAt?: string;

  inviterId: string;
  inviter: User;

  /** @deprecated Use 'status' instead */
  isAccepted: boolean;
}
